_test/interface40.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import "fmt"
+
+type foo struct {
+	bar string
+}
+
+func (f foo) String() string {
+	return "Hello from " + f.bar
+}
+
+func NewFoo(s string) fmt.Stringer {
```

Comment:
By convention, a function prefixed with "New" usually returns a pointer. So the "problem" here is two-fold: 1) the result in the signature is an interface, not a pointer, and 2) a foo is returned, not a pointer to a foo.
I would suggest s/NewFoo/Foo/ for example.

Reply:
ok, done